testtools/files.go

```diff
@@ -129,3 +139,232 @@ func CheckFiles(t *testing.T, dir string, files []FileSpec) {
 		}
 	}
 }
+
+type TestGazelleGenerationArgs struct {
+	// name is the name of the test.
+	Name string
+	// testDataPath is the workspace relative path to the test data directory.
+	TestDataPath string
+	// gazelleBinaryDir is the workspace relative path to the location of the gazelle binary
+	// we want to test.
+	GazelleBinaryDir string
+	// gazelleBinaryName is the name of the gazelle binary target that we want to test.
+	GazelleBinaryName string
+
+	// The suffix for all test input build files. Includes the ".".
+	// Default: ".in", so input BUILD files should be named BUILD.in.
+	BuildInSuffix string
+
+	// The suffix for all test output build files. Includes the ".".
+	// Default: ".out", so out BUILD files should be named BUILD.out.
+	BuildOutSuffix string
+}
+
+func NewTestGazelleGenerationArgs() *TestGazelleGenerationArgs {
+	return &TestGazelleGenerationArgs{
+		BuildInSuffix:  ".in",
+		BuildOutSuffix: ".out",
+	}
+}
+
+// TestGazelleGenerationOnPath runs a full gazelle binary on a testdata directory.
+// With a test data directory of the form:
+//└── <testDataPath>
+//    └── some_test
+//        ├── WORKSPACE
+//        ├── README.md --> README describing what the test does.
+//        ├── test.yaml --> YAML file for test configuration.
+//        └── app
+//            └── sourceFile.foo
+//            └── BUILD.in --> BUILD file prior to running gazelle.
+//            └── BUILD.out --> BUILD file expected after running gazelle.
+func TestGazelleGenerationOnPath(t *testing.T, args *TestGazelleGenerationArgs, files []bazel.RunfileEntry) {
```

Comment:
Can we get away with out taking in the set of `files`? If the `TestGazelleGenerationArgs` has the `TestDataPath` I believe we can get all of these from listing the files on disk and make the interface more robust.